File: examples/python/octra_ex.py
from pyoctra import octra

# dir(octra)

octra.hello()

dp = octra.DPair(1, 2)

s = octra.SVector(3)
s[0] = "1"
s[1] = "2"
s[2] = "3"
# s[3] = "4"  # <- Gets index out of bounds error


class TimesTwo(octra.Callback):
    def call(self, x):
        return x * 2.0


class TimesThree(octra.Callback):
    def call(self, x):
        return x * 3.0


# cb = TimesTwo()
cb = TimesThree()
print("call_with_callback(3.0) =", octra.call_with_callback(3.0, cb))
v = octra.make_dvector(1.0, 2.0, 3.0)
v2 = octra.map_dvector_with_callback(v, cb)
print("sum_dvector(map_dvector_with_callback(1,2,3)) =", octra.sum_dvector(v2))

File: examples/ruby/octra_ex.rb
require "octruby"

Octra.hello

dp = Octra::DPair.new(1.0, 2.0)
puts "DPair created: #{dp.inspect}"
dp.delete if dp.respond_to?(:delete)

class TimesTwo < Octra::Callback
  def call(x)
    x * 2.0
  end
end

cb = TimesTwo.new
puts "call_with_callback(3.0) = #{Octra.call_with_callback(3.0, cb)}"
v = Octra.make_dvector(1.0, 2.0, 3.0)
v2 = Octra.map_dvector_with_callback(v, cb)
puts "sum_dvector(map_dvector_with_callback(1,2,3)) = #{Octra.sum_dvector(v2)}"
